### ISSUE TITLE
malloc results don't seem aligned.

### ISSUE THREAD

**Issue by glommer** [jjyr/buddy-alloc#5]
I have allocated aligned memory to use with this:

```rust
        let data = aligned_alloc(size, 4 << 10) as *mut u8;
        println!("Buffer data: {:p}", data);
```
prints:
```
$ Buffer data: 0x7f0a935ff000
```

And initialized the `BuddyAlloc` as follows:
```rust
        let allocator = unsafe { BuddyAlloc::new(BuddyAllocParam::new(data.as_ptr(), size, 4096)) };
```

But when I call `malloc`, The address that is returned is:

```
0x7f0a93ffd80c
```

Usually in Buddy implementations like the Linux Kernel one wants to use this to allocate things resembling pages, which have a natural alignment.

My use case for this is to implement an allocator for the `io_uring` Direct I/O registered buffers for using with the [https://github.com/DataDog/glommio](glommio) crate. However Direct I/O imposes page-alignment and I need every buffer that comes out of this to be aligned to the block size.

Is there any way such guarantee can be provided ?

**Comment by jjyr** [comment]
I can't reproduce this issue in the latest version. Can you provide a minimal executable example to reproduce it?

**Comment by glommer** [comment]
What is the latest version ? git? I am using v0.4.0 from crates.

I will work on a reproducer right now

**Comment by jjyr** [comment]
v0.4.0 is the latest version.

**Comment by glommer** [comment]
The following fails for me in the second assert (meaning the buffer itself is aligned , first assert)
```rust
    #[test]
    fn allocator() {
        use aligned_alloc::{aligned_alloc, aligned_free};

        let size = 16 << 20;
        let data = aligned_alloc(size, 4 << 10) as *mut u8;
        assert_eq!(data as usize & 4095, 0);
        let data = std::ptr::NonNull::new(data).unwrap();
        let mut allocator = unsafe { BuddyAlloc::new(BuddyAllocParam::new(data.as_ptr(), size, 4096)) };
        let x = allocator.malloc(4096);
        assert_eq!(x as usize & 4095, 0);
    }
```

My cargo has:
```
aligned_alloc = "0.1"
buddy-alloc = "0.4.0"
```

**Comment by jjyr** [comment]
I see, there is one line missed to align the base_addr https://github.com/jjyr/buddy-alloc/commit/90c7a22c158ca0cf73c8d80af00a212bb828269a#diff-c8d1e664fcb6f473f946eafbeab39833216c9d43e7eb0e53f8c507b1ba080e04R243

Fixed in the v0.4.1.

**Comment by glommer** [comment]
Appreciate the quick fix!
